tools/convert.py

```diff
@@ -120,6 +125,11 @@ def handle_tensors(args, writer, state_dict):
                 #data = np.squeeze(data) # don't do this
                 data_qtype = gguf.GGMLQuantizationType.F16
 
+        if data_qtype != gguf.GGMLQuantizationType.F32 and n_dims >= 2 and n_params >= 256*256 and (n_params / 256).is_integer() and data.shape[0] >= 256 and data.shape[1] >= 256:

```

Comment:
I think the n_dims == 4 check for FP16 above this can safely be removed since we're not doing any quantization in this script (i.e. not used anymore).

Reply:
this heuristic should be adjusted also, 256*256 is probably to small to be worth quantizing. also it's only checking if we _can_ rearrange stuff but not if we must, so it's probably doing that more aggressively than is actually necessary.